### PR TITLE
SV-402 Remove ID from inner Api, use Lars Code when translating

### PR DESCRIPTION
**Pull request by CodeWolf86**

### CODE REVIEW

File: src/SFA.DAS.Campaign.Api.UnitTests/Controllers/TrainingCourses/WhenGettingStandardsBySector.cs [hunk 1]
```diff
@@ -37,13 +38,14 @@ public class WhenGettingStandardsBySector
             controllerResult.StatusCode.Should().Be((int)HttpStatusCode.OK);
             var model = controllerResult.Value as GetStandardsResponse;
             Assert.IsNotNull(model);
-            model.Standards.Should().BeEquivalentTo(mediatorResult.Standards, options => options);
+            model.Standards.Should().BeEquivalentTo(mediatorResult.Standards, options => options.ExcludingMissingMembers());
```
Comment (DevenKShah) [inline src/SFA.DAS.Campaign.Api.UnitTests/Controllers/TrainingCourses/WhenGettingStandardsBySector.cs:41]: `ExcludingMissingMembers` is a good utility, however I think we should exclusively list members so when new members are added, the test fails and gives developer an indication of impact, against which they can take conscious actions.

File: src/SFA.DAS.Campaign/InnerApi/Responses/GetStandardsListItem.cs [hunk 2]
```diff
@@ -2,6 +2,8 @@
 {
     public class GetStandardsListItem 
     {
-        public int Id { get; set; }
+        public string StandardUId { get; set; }
+        public int LarsCode { get; set; }
+
```
Comment (DevenKShah) [inline src/SFA.DAS.Campaign/InnerApi/Responses/GetStandardsListItem.cs:7]: 👾

File: src/SFA.DAS.Approvals.Api.UnitTests/Models/WhenMappingGetStandardResponseFromMediatorResponse.cs [hunk 3]
```diff
@@ -19,11 +19,14 @@ public void Then_The_Fields_Are_Mapped(GetStandardsListItem source)
...
                 .Excluding(c=>c.TypicalDuration)
                 .Excluding(c=>c.IsActive)
+                .Excluding(c => c.StandardUId)
+                .Excluding(c => c.LarsCode)
             );
             actual.EffectiveFrom.Should().Be(source.StandardDates.EffectiveFrom);
             actual.EffectiveTo.Should().Be(source.StandardDates.EffectiveTo);
             actual.LastDateForNewStarts.Should().Be(source.StandardDates.LastDateStarts);
             actual.Duration.Should().Be(source.TypicalDuration);
+            actual.Id.Should().Be(source.LarsCode);
```
Comment (DevenKShah) [inline src/SFA.DAS.Approvals.Api.UnitTests/Models/WhenMappingGetStandardResponseFromMediatorResponse.cs:30]: 👾

File: src/SFA.DAS.FindEpao.Api.UnitTests/Controllers/Courses/WhenGettingCourseList.cs [hunk 4]
```diff
@@ -35,7 +36,8 @@ public class WhenGettingCourseList
             controllerResult.StatusCode.Should().Be((int)HttpStatusCode.OK);
             var model = controllerResult.Value as GetCourseListResponse;
             Assert.IsNotNull(model);
-            model.Courses.Should().BeEquivalentTo(mediatorResult.Courses);
+            model.Courses.Should().BeEquivalentTo(mediatorResult.Courses, options => options.ExcludingMissingMembers());
```
Comment (DevenKShah) [inline src/SFA.DAS.FindEpao.Api.UnitTests/Controllers/Courses/WhenGettingCourseList.cs:39]: See previous comment on `ExcludingMissingMembers`

File: src/SFA.DAS.FindEpao.Api.UnitTests/Models/WhenCastingGetTrainingCoursesResponseFromMediatorType.cs [hunk 5]
```diff
@@ -14,7 +14,8 @@ public class WhenCastingGetTrainingCoursesResponseFromMediatorType
         {
             var response = (GetCourseListItem)source;
 
-            response.Should().BeEquivalentTo(source);
+            response.Should().BeEquivalentTo(source, options => options.ExcludingMissingMembers());
+            response.Id.Should().Be(source.LarsCode);
         }
```
Comment (DevenKShah) [inline src/SFA.DAS.FindEpao.Api.UnitTests/Models/WhenCastingGetTrainingCoursesResponseFromMediatorType.cs:20]: 👾

File: src/SFA.DAS.Forecasting.Api.UnitTests/Models/WhenCastingApprenticeshipCourseFromMediatorType.cs [hunk 6]
```diff
@@ -23,11 +23,13 @@ public class WhenCastingApprenticeshipCourseFromMediatorType
...
                     .Excluding(s => s.StandardDates)
                     .Excluding(s => s.IsActive)
+                    .Excluding(s => s.StandardUId)
+                    .Excluding(s => s.LarsCode)
                 );
 
             response.Duration.Should().Be(source.TypicalDuration);
             response.FundingCap.Should().Be(source.MaxFunding);
-            
+            response.Id.Should().Be(source.LarsCode.ToString());
```
Comment (DevenKShah) [inline src/SFA.DAS.Forecasting.Api.UnitTests/Models/WhenCastingApprenticeshipCourseFromMediatorType.cs:32]: Was this wrongly returning a string type on Id?

File: src/SFA.DAS.Forecasting.Api/Models/ApprenticeshipCourse.cs [hunk 7]
```diff
@@ -20,7 +20,7 @@ public class ApprenticeshipCourse
         {
             return new ApprenticeshipCourse
             {
-                Id = standard.Id,
+                Id = standard.LarsCode.ToString(),
```
Comment (DevenKShah) [inline src/SFA.DAS.Forecasting.Api/Models/ApprenticeshipCourse.cs:23]: 🤔

File: src/SFA.DAS.Reservations.Api.UnitTests/Controllers/WhenCallingGetTrainingCourseList.cs [hunk 8]
```diff
@@ -36,7 +37,8 @@ public class WhenCallingGetTrainingCourseList
             controllerResult.StatusCode.Should().Be((int)HttpStatusCode.OK);
             var model = controllerResult.Value as GetTrainingCoursesListResponse;
             Assert.IsNotNull(model);
-            model.Standards.Should().BeEquivalentTo(mediatorResult.Courses);
+            model.Standards.Should().BeEquivalentTo(mediatorResult.Courses, options => options.ExcludingMissingMembers());
```
Comment (DevenKShah) [inline src/SFA.DAS.Reservations.Api.UnitTests/Controllers/WhenCallingGetTrainingCourseList.cs:40]: ☝️

File: src/SFA.DAS.Reservations.Api.UnitTests/Models/WhenCastingGetTrainingCoursesResponseFromMediatorType.cs [hunk 9]
```diff
@@ -14,7 +14,8 @@ public class WhenCastingGetTrainingCoursesResponseFromMediatorType
         {
             var response = (GetTrainingCoursesListItem)source;
 
-            response.Should().BeEquivalentTo(source);
+            response.Should().BeEquivalentTo(source, options => options.ExcludingMissingMembers());
```
Comment (DevenKShah) [inline src/SFA.DAS.Reservations.Api.UnitTests/Models/WhenCastingGetTrainingCoursesResponseFromMediatorType.cs:17]: ☝️

**Review by DevenKShah — APPROVED** [review]
This looks good. Just one comment on use of `ExcludingMissingMembers`. 

I think we will have to create a SV main branch here as well so we can accumulate our changes on that branch till we decide to go live.

File: src/SFA.DAS.Campaign/InnerApi/Responses/GetStandardsListItem.cs (hunk 2 above)
Comment (CodeWolf86) [reply src/SFA.DAS.Campaign/InnerApi/Responses/GetStandardsListItem.cs:7]: 🚀

File: src/SFA.DAS.Approvals.Api.UnitTests/Models/WhenMappingGetStandardResponseFromMediatorResponse.cs (hunk 3 above)
Comment (CodeWolf86) [reply src/SFA.DAS.Approvals.Api.UnitTests/Models/WhenMappingGetStandardResponseFromMediatorResponse.cs:30]: 🚀

File: src/SFA.DAS.FindEpao.Api.UnitTests/Models/WhenCastingGetTrainingCoursesResponseFromMediatorType.cs (hunk 5 above)
Comment (CodeWolf86) [reply src/SFA.DAS.FindEpao.Api.UnitTests/Models/WhenCastingGetTrainingCoursesResponseFromMediatorType.cs:20]: 🚀

File: src/SFA.DAS.Forecasting.Api.UnitTests/Models/WhenCastingApprenticeshipCourseFromMediatorType.cs (hunk 6 above)
Comment (CodeWolf86) [reply src/SFA.DAS.Forecasting.Api.UnitTests/Models/WhenCastingApprenticeshipCourseFromMediatorType.cs:32]: No I had a look at this wondering if it was a mistake. The `ApprenticeshipCourse` object can be populated by either a Framework or a Standard, The ID of a Framework is a string, so due to the re-use of an object, both the StandardId and FrameworkId fit into the string Id field so I followed suit.

File: src/SFA.DAS.Forecasting.Api/Models/ApprenticeshipCourse.cs (hunk 7 above)
Comment (CodeWolf86) [reply src/SFA.DAS.Forecasting.Api/Models/ApprenticeshipCourse.cs:23]: See above ☝️

File: src/SFA.DAS.Campaign.Api.UnitTests/Controllers/TrainingCourses/WhenGettingStandardsBySector.cs (hunk 1 above)
Comment (CodeWolf86) [reply src/SFA.DAS.Campaign.Api.UnitTests/Controllers/TrainingCourses/WhenGettingStandardsBySector.cs:41]: I've switched to specifics on `StandardUId` and `LarsCode` I don't like either, but at least as you say this is explicit.

File: src/SFA.DAS.FindEpao.Api.UnitTests/Controllers/Courses/WhenGettingCourseList.cs (hunk 4 above)
Comment (CodeWolf86) [reply src/SFA.DAS.FindEpao.Api.UnitTests/Controllers/Courses/WhenGettingCourseList.cs:39]: As above, switched to explicits.

File: src/SFA.DAS.Reservations.Api.UnitTests/Controllers/WhenCallingGetTrainingCourseList.cs (hunk 8 above)
Comment (CodeWolf86) [reply src/SFA.DAS.Reservations.Api.UnitTests/Controllers/WhenCallingGetTrainingCourseList.cs:40]: As above, switched to explicits.

File: src/SFA.DAS.Reservations.Api.UnitTests/Models/WhenCastingGetTrainingCoursesResponseFromMediatorType.cs (hunk 9 above)
Comment (CodeWolf86) [reply src/SFA.DAS.Reservations.Api.UnitTests/Models/WhenCastingGetTrainingCoursesResponseFromMediatorType.cs:17]: As above switched to explicits

**Review by alandburns — APPROVED** [review]

**Comment by DevenKShah** [comment]
This changes now already merged in SV-170

**Pull request closed without merging**